presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcPageSourceFactory.java

```diff
@@ -257,7 +259,7 @@ private static OrcPageSource createOrcPageSource(
             }
 
             Map<String, OrcColumn> fileColumnsByName = ImmutableMap.of();
-            if (useOrcColumnNames || isFullAcid) {
+            if (useOrcColumnNames || (isFullAcid && !originalFilesPresent)) {
```

Comment:
we should verify expected schema also when handling original files

Reply:
fixed.